bigquery/src/main/scala/com/yahoo/maha/executor/bigquery/BigqueryQueryExecutor.scala

```diff
@@ -0,0 +1,174 @@
+  gcpCredentialsFilePath: String,
+  gcpProjectId: String,
+  enableProxy: Boolean,
+  proxyCredentialsFilePath: Option[String],
+  proxyHost: Option[String],
+  proxyPort: Option[String],
+  retries: Int
+) extends Logging {
+
+  private final val HttpsProxyHost = "https.proxyHost"
+  private final val HttpsProxyPort = "https.proxyPort"
+  private final val DisabledAuthSchemes = "jdk.http.auth.tunneling.disabledSchemes"
+
+  private val mapper = {
+    val om = new ObjectMapper(new YAMLFactory()) with ScalaObjectMapper
+    om.registerModule(DefaultScalaModule)
+    om.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+    om
+  }
+
+  def buildBigqueryClient(): BigQuery = {
+    BigQueryOptions
+      .newBuilder()
+      .setCredentials(ServiceAccountCredentials.fromStream(new FileInputStream(gcpCredentialsFilePath)))
+      .setProjectId(gcpProjectId)
+      .setRetrySettings(ServiceOptions.getDefaultRetrySettings().toBuilder().setMaxAttempts(retries).build())
+      .build()
+      .getService
+  }
+
+  def configureProxy(): Unit = {
+    if (enableProxy && proxyCredentialsFilePath.isDefined && proxyHost.isDefined && proxyHost.isDefined) {
+      val proxyCredentials = mapper.readValue[ProxyCredentials](
+        new FileInputStream(proxyCredentialsFilePath.get)
+      )
+
+      System.setProperty(HttpsProxyHost, proxyHost.get)
+      System.setProperty(HttpsProxyPort, proxyPort.get)
+      System.setProperty(DisabledAuthSchemes, "")
+
+      Authenticator.setDefault(new Authenticator() {
+        override protected def getPasswordAuthentication =
+          new PasswordAuthentication(proxyCredentials.userName, proxyCredentials.password.toCharArray)
+      })
+    } else {
+      info("Not using proxy for BigQuery Query Execution")
+    }
+  }
+}
+
+class BigqueryQueryExecutor(
+  config: BigqueryQueryExecutorConfig,
+  lifecycleListener: ExecutionLifecycleListener
+) extends QueryExecutor
+  with Logging {
+
+  val engine: Engine = BigqueryEngine
+  config.configureProxy()
+  lazy val bigqueryClient = config.buildBigqueryClient()
+
+  def execute[T <: RowList](
+    query: Query,
+    rowList: T,
+    queryAttributes: QueryAttributes
+  ): QueryResult[T] = {
+    val acquiredQueryAttributes = lifecycleListener.acquired(query, queryAttributes)
+    val debugEnabled = query.queryContext.requestModel.isDebugEnabled
+
+    if (!acceptEngine(query.engine)) {
+      throw new UnsupportedOperationException(
+        s"BigqueryQueryExecutor does not support query with engine=${query.engine}"
+      )
+    }
+
+    val formattedQuery = SqlFormatter.format(
+      query.asString,
+      FormatConfig(dialect = SqlDialect.StandardSQL)
+    )
+
+    if (debugEnabled) {
+      info(s"Running query : $formattedQuery")
+    }
+
+    val queryJobConfig = QueryJobConfiguration
+      .newBuilder(formattedQuery)
+      .setUseLegacySql(false)
+      .build()
+
+    val jobId = JobId.of(UUID.randomUUID.toString)
```

Comment:
Could you reuse the request id here ?  Would that make sense for tracking or do you want this to be different ?

Reply:
Switched over to request ID